app/assets/javascripts/chart-data-mixin.js

```diff
@@ -0,0 +1,129 @@
+angular.module('containerDashboard').factory('ChartsDataMixin', ['$timeout', '$q', function chartDataMixinFactory ($timeout, $q) {
```

Comment:
> @abonas As I'm trying to integrate the 'Node utilization' cards I'm trying to figure out if container_dashboard_service.rb should do most of these and give the data ready to be inserted to the card. Your thoughts?

good direction. the calculation should happen in the service, then the binding/injection in the controller